src/main/scala/wdlTools/types/TypeInfer.scala

```diff
@@ -308,15 +330,19 @@ case class TypeInfer(conf: TypeOptions, errorHandler: Option[Vector[TypeError] =
         TAT.ExprMap(Map.empty, T_Map(T_Any, T_Any), text)
 
       case AST.ExprMap(value, text) =>
-        // figure out the types from the first element
         val m: Map[TAT.Expr, TAT.Expr] = value.map { item: AST.ExprMapItem =>
           applyExpr(item.key, bindings, ctx) -> applyExpr(item.value, bindings, ctx)
         }.toMap
-        // unify the key types
-        val tk = unifyTypes(m.keys.map(_.wdlType), "map keys", text, ctx)
-        // unify the value types
-        val tv = unifyTypes(m.values.map(_.wdlType), "map values", text, ctx)
-        TAT.ExprMap(m, T_Map(tk, tv), text)
+        val wdlType = if (unifyMaps) {
```

Comment:
Are we still still using unifyMaps? Is there a case where don't have one key type, and one value type?

Reply:
oops - just forgot to remove it. fixed now. please re-review